### DISCUSSION TITLE
My feeble attempts...

### DISCUSSION THREAD

**Discussion by kurotatsu77** [IHateMyKite/UnforgivingDevices#110] [General]
So, after some discussion I've tried to do various things, to learn while doing, and I think I have something which might serve as basis for the UD quest mod, which was deemed to be as separate esp or some material for inclusion in to to said mod and/or UD itself.

Repository for you to have a look at: https://github.com/kurotatsu77/UnforgivingSkyrim

My suggestion is to stick with DD5.2 Beta and develop only for it, should reduce support hassle and we'll have mod all good to release upon DD 5.2 release.

Here's questions I have:
1. Do you think it will be good basis to build quest mod for UD?
2. Do you have any pointers for me, especially for known problems I've written in the readme?
3. Do you like UD quest basically being this: PC learns Black Goo first effect, after it option appears in dialogue with alchemists to ask about it, all of them would suggest PC to talk with Arcadia in Whiterun. We will show this to her, some bondage shenanigans can ensue (we can either offer her this to try herself or go with her suggestion to demonstrate it), after which she will promise to contact us with information. After 24 hours courier arrives with letter from the new NPC we create - for now I name her Nodaba The Seeker. She will offer to meet us at Bannered Mare Inn in the Whiterun to discuss researching the Black Goo. After meeting her (and maybe again demonstrating effect, because more bondage! :) ) she would suggest buying home in Whiterun so we can have proper lab to work. After completing all home upgrades we get option to make basement to Breezehome, basically adding small trapdoor either outside near Breezehome and/or inside it. This basement cell can be made any size, theme and whatever else to suit our needs and should provide minimal clash with all other mods. After building we go to our helper and she moves into the basement, and the main cycle starts, where we expand the lab and open recipes (which are basically available from the start, but it's easy to override). This will provide us with money and goo sink into the research, plus she can give us side quests. Couple of those I have in mind: bring in the chastity devices book from the College and another being to learn Abadon Shout. Current Abadon Curse quest slides into this quite nicely with no obvious conflict. Maybe later add some simple radiant ones (repeatable, bondage escape challenges, bring some goo/whatever, ...).

With #3 a lot of input and suggestions would be appreciated greatly!

**Comment by iiw2012** [comment]
> Do you think it will be good basis to build quest mod for UD?

It's great as long as you like it!

> Do you have any pointers for me, especially for known problems I've written in the readme?

Abadon Slave Harness uses biped slot 49 that is reserved for belts I suppose.

![image](https://user-images.githubusercontent.com/110040088/212466492-b6ee454f-560f-4fc9-bee3-249895b0ede9.png)

And models are messed up in Abadon Belt overrides.

![image](https://user-images.githubusercontent.com/110040088/212466550-563f2234-dc47-40d6-bc3e-5961c59d93a6.png)

> Do you like UD quest basically being this: ...

I like it! Enough with bondage fun from random traps, it's time to add more plot and compulsion. :)

**Comment by kurotatsu77** [comment]
Thank you so much! I'll take a look at the slots thing!

Models are messed because you don't have DD 5.2 Beta 10 installed, I've made override to use Black Lustrous look from it. It blends into black ebonite look so PERFECTLY! :)

**Comment by IHateMyKite** [comment]
Will give it a look once I have more time. But regarding the 3. question, I think its great. I don't see anything impossible about it. 
Hardest point might be to add the trapdoor to the Breeze Home, as if some user have mod which rework the Breeze Home, it will be incompatible. I had similar problems before when making More Devious Starts. Best might be to just add new house somewhere in wilderness which you would have to buy from someone. Let me know what you think.

**Comment by kurotatsu77** [comment]
Hmm, actually there could be nice option... You have created the house which is used in the LAL addon in bound at home scenario.
Was it proven to be conflict-free? Maybe use this, add some checks to see if that option was used as start, just a small branch like "Oh, I have a nice house nearby, we can use it." And if it proves to be too small - we can add basement expansion with trapdoor to it.

We can use it without adding hard dependency on the LAL addon, yeah?

**Reply by IHateMyKite** [reply]
I didn't get any complaints about the house causing issues with other mods, so I might have struck the gold when placing it. But making it dependant on LAL is not very good idea, as the LAL can't be even downloaded anymore. But making new house is not very hard. You just copy some existing one. This is what I did, I have used one house from Whiterun, and deleted some inside object. I also had to update the doors, as when you copy the house, they will still link to the previous house. 
I think that locations north from Whiterun are not used that much, so placing the house there would most likely be no issue.

**Comment by kurotatsu77** [comment]
This is great! Good placement, near Whiterun, but separated, great for naughty research. It can cost a lot less than Breezehome because out of city walls. 
Now all that left is to figure out if we can use the same house in the LAL addon and US (this is how I call this quest mod).
Or maybe try to find another spot nearby and do the house copy thing...

**Comment by kurotatsu77** [comment]
Just a small heads-up - I've finished the quest to obtain the Abadon Shout. Learned a lot about quests and dialogues while was doing it, will be very useful in future. 
For now can you all please have a look at it and tell me what you think about it?
Meanwhile I'll start reading up and trying to learn how to add houses and NPCs...

**Reply by IHateMyKite** [reply]
Great, will test it. Regarding the house, I have already started working on it some time ago (created side branch for it). The house is placed north east of whiterun. An issue is that I need to correct the terrain around the house, which is a big pain in the ass, because if you fuck it up, you can't undo it in CK, you just have to reload the mod. Once I'm done, I will create PR.

**Reply by iiw2012** [reply]
Started a new game and went to Bleak Falls Barrow. Got hit by Abadon Shout from the last boss. Got a heavy binding, but no quest  started. Is this intended behavior?

I noticed some conventions when traveling with a companion. The fight stops for the whole group if at least one gets hit from shout. Maybe remove only the victim from the combat?

**Reply by kurotatsu77** [reply]
Quest won't start until MQ104 finished ("Dragons Rising" where you Shout first time after killing the first dragon). Also it will start if follower hit with Abadon Shout and not player. But! MQ104 has to be complete, so player knows about being Dragonborn and being able to Shout.

In fact script stop combat for victim and assaulter (Caster and Target), no interaction for others. My guess - others follow suit automatically.

**Reply by iiw2012** [reply]
> Quest won't start until MQ104 finished ("Dragons Rising" where you Shout first time after killing the first dragon).

I moved along the plot and got acquainted with the Abadon Shout before MQ104 quest. It might be worth adding a separate stage for the Abadon Shout quest, so that the player can at least note in the journal strange abilities of the boss-draugrs since they use it.

Something like that:
`
I came across a strange draugr in one of the dungeons. After his sonic boom devilish immobilizing devices appeared on me. And the draugr, instead of finishing me off, stood and watched my attempts to get out! I've never paid attention to their abilities before. Can a living person master them?
`

`
I came across a strange draugr in one of the dungeons. After his sonic boom devilish immobilizing devices appeared on my companion. And the draugr, instead of finishing us off, stood and watched our attempts to get out! I've never paid attention to their abilities before. Can a living person master them?
`

**Reply by kurotatsu77** [reply]
Hmmm, interesting idea, I'll give it a thought...

**Reply by kurotatsu77** [reply]
OK, added these two stages, had to cut down log strings a bit to fit into the limitation.
To advance to actually learning first Word player has to witness the Shout once more, kind of makes sense, I guess?

**Comment by kurotatsu77** [comment]
Small progress update on the quest development:
Made the quest, it starts when visiting Whiterun area while player knows first effect of black goo and has at least one in inventory.
Created NPC, placed her at Bannered Mare Inn, made stages of talking to Arcadia and Nodaba (name of the NPC). Scene plays out as it should. It marks the stage when we can actually start the research main loop, all introductory stuff is done. Only one thing is not working is package which should send Nodaba to the house, nothing works, including really forced ways of adding the package. My suspicion is problematic navmesh, which I can't fix because of the goddamn CK keeps crashing when I try to open more or less big cell into edit, no tweaks I've found on Inet were able to fix this so far. Unfortunately, I do need to be able to edit this, to add research activators inside the house or ask for anyone willing to help me.

If anyone takes a look into this and offers some help/advice/feedback - it would be welcome.

**Reply by iiw2012** [reply]
I'll see what I can do. Need some time to test it.

**Reply by iiw2012** [reply]
Okay, I think I figured out the problem.

Looks like CK crashes if you have too much installed animations. I have to disable mod with FNIS output while working in CK and cells are loaded without problems. Or you could reduce the number of SLAL packs.

As for the Nodaba package I've found several issues that prevent package from playing:
- NavMesh near the Hub was not adapted to the building and its doors.
- Door teleports was invalid.
- NavMesh inside building was not finalized with the door teleport.
- Area inside the inn forces actors to play special sandbox package (i have no idea why and how). So I've changed package US_NodabaToHub to force Nodaba out of the tavern and then her native sandbox packages will bring her to the Hub.

All these fixes along with some not related ones is in the branch https://github.com/kurotatsu77/UnforgivingSkyrim/tree/dev-goo-research-quest-01

**Reply by kurotatsu77** [reply]
Thank you very much!
What's your general impression? Did you liked how it was all set up?
Any suggestions?

**Reply by kurotatsu77** [reply]
This was brilliant, absolutely loving your touches on the scene! Thank you very much!

One little snag was only esp having Form 44 and CTDing my LE on game load. Resolved it by resaving it LE CK, probably will need testing to check if it's still properly working in SE/AE.

**Reply by iiw2012** [reply]
> One little snag was only esp having Form 44 and CTDing my LE on game load.

Sorry about that. I'll be more carefull in the future with my SSE CK.

> What's your general impression? Did you liked how it was all set up?

I really liked the dialogues. And a plot twist with the house.

You can also add a couple of passages at the beginning that BlackGoo allows you to comprehend the alchemy of Dragons (or something like that). Otherwise, the player's motivation to throw himself into the "caring" hands of Nobada comes down to simple masochism :).

Let the player aquire a book or something? Probably you've planned that for later :)

> Any suggestions?

BlackGoo and even more so ConcentratedBlackGoo is a serious challenge, especially at early levels, when the player may not even have enough lockpicks (or patience).

I do not know what can replace these tests (it depends on further progression). But you can provide ways to quickly release the player. For example, Arcadia can help the player get free after completing some simple task (sexual or not).

Or let the player not be exempt from all devices before scene with Nodaba on stage 20.

**Reply by kurotatsu77** [reply]
These are very solid suggestions and nice food for thought. I do entertain idea of player embracing masochist side along the lines, with Nodaba getting off on sight of bound, struggling and whipped player :D And I want player to struggle a lot to get out. Also there are plenty of other mods, DCL, for example, which offer some way out. Also UD got nice help dialogue for NPCs, which also plays good into it. Getting much more rich conversation with Arcadia to ignite interest and with Nodaba (and maybe with some dare like "are you that weak to be afraid of this?") for more incentive to go along with her orders is a very, very good idea. Unfortunately, my writing skills are quite mediocre... Food for thought and improvement! Again, thank you very much!

**Comment by kurotatsu77** [comment]
Another update, great news! Thanks to invaluable advice of @iiw2012 I finally managed to make my CK to work properly in 3D view, which allowed me to work on the completing basic part of the Goo Research Quest. Also figured out why esp was mangled by CK before.

Now almost all recipes are locked behind the research, which conducted simply by throwing materials and gold into the chest, Nodaba tells details. Research points are accumulated with time. Placed disabled alchemy and enchanting stations which would boost research, thus allowing for side quests for obtaining them to be created. In addition to researching items from UD, it also possible now to learn effects of generic ingredients too, as long as they have VendorIngredient keyword.

Please try it out and tell me your impressions. And with this foundation in place you are welcome to throw in some fun quests and other additions :)

**Reply by IHateMyKite** [reply]
Testes up to the latest version, and it's superb. Didn't research all recipes yet, but so for I didn't find issue. Regarding the questions, how exactly do you buy an alchemy/enchanting table ?

Regarding the ideas, I would love to add some more NPC interactions with bondage. As we discussed some time ago, adding some kind of rescue quest would be great. I will have to first update the framework tho, as there will be need for some kind of mod specific NPC manager (so every mod can have their own NPC slots, which can be used for quest NPC). Once it's done, it will allow to dome so cool things, like the rescue quest. 
Another idea (related to previous idea) would be that there would be small chance for research to fail, which will cause Nodaba to be locked in devices. This will prompt a side quest which will tell the player to return to the house and help her out. Player can ignore it, in which can she have to help her self (hopefully AI will work as intended). But by helping her, there might be some kind of reward (bonus research points, discount on research or just money).

**Reply by kurotatsu77** [reply]
Thank you for taking a look at it and so quickly! Your praise means a lot to me! :)

To buy stations - it's via two side quests, now in planning phase. So far I'm thinking Arcadia for alchemy, which would offer boring option of simply paying her up (and Nodaba scolding player for wasting money afterwards) and more fun, bondagey and sexy options.

Enchanting should be done via another side quest, but to whom we should send player? I'm so far thinking Cancelmo in Markarth, but completely at loss what naughty things can be placed in it.

Finally I have idea of side quest to get that good old Devious Devices Book from college, and Urag was naughty on in the original DD quest, from which I take inspiration. Maybe even remake forbidden bookcase into door to another small playcell of Urag? Ideas are many!

And you probably can guess that Hulda has something for player to do, repeatable quest to perform bondage entertainment of patrons in inn.

Idea about saving NPC as radiant quest is awesome, I like it! 

Also I had idea of getting orders to punish bad girls by coming to them, binding and leaving them locked at public place (maybe inn?) in stocks or other furniture to humiliate them. And imagine choices player would have if one day she would receive order to punish herself! :D Boring option would be to pay out bounty to the client, maybe with a bit of extra. Fun options would be going through it or even offer more harsh treatment, because gold is good! :D

Idea about Nodaba having accidents is very fun, I like it!

Now one problem though... So many fun ideas, so many fun quests to make, too low on time! :)

Also current implementation is a bit sceletal, which was intentional. I needed to have some foundation to build upon. Now some "meat" can be added to it, expanding dialogues and lore, @iiw2012 made quite a number of good suggestions about it.

But with this step complete I do feel like we can do it now, one quest at time, slow and steady. If you or any other wants to work on one of these (or other) quests - please send heads-ups, so we coordinate our work and don't waste efforts into doing the same thing.

**Comment by kurotatsu77** [comment]
Quest to get Alchemy workstation from Arcadia and her repeatable action is done.
Feel free to give it a spin.

On other quests work is on hold for now because of lack of clear vision how I want them to be implemented in details. Thinking and planning time!

If any of you decide to try to implement one of those quests - please advise to avoid double-effort.

**Comment by iiw2012** [comment]
I've been testing UD time after time, and an idea came to my mind:
What if we add a mini-game to collect BlackGoo from plants, veins (chests?). Like this:
1. The player tries to collect a flower.
2. A message appears that something is stuck on the flower. With options: ignore and pick a flower, try to collect sticky muck or run away.
2.1. If "ignore" is selected, then there is a chance:
2.1.1. Get a flower.
2.1.2. Get a flower and get smeared with BlackGoo.
2.2. If "gather sticky muck" is selected, then a mini-game starts similar to UD mini-games (with an Alchemy skill check).
2.2.1. An appropriate animation begins.
2.2.2. If the player hits a crit, he gets BlackGoo.
2.2.3. If the player misses, he is smeared with BlackGoo and the game ends.

The more possibilities that the player will be in bondage against their will (or because of their mistake), the more interesting.

I really want to try to implement it myself, but it seems that I have already picked up more tasks than I have time to do :( 
I'll try to take it up, but a little later. In any case, if necessary, I will make animations for the player tries to cope with the sticky muck.

**Reply by kurotatsu77** [reply]
It looks like a fun idea! But yeah, implementing could be a bit of hassle. And possible DCL intervention can be predicted here.

Another thought that was nagging me was that there's no lasting consequences to losing mind while wearing 100% charged Abadon Plug.
But still, I wasn't been able to think up some nice curse to be used in it...

**Reply by iiw2012** [reply]
> And possible DCL intervention can be predicted here.

As far as I remember, DCL works through perk on activate action. So it is possible to add own perk and count interactions with plants, veins, etc. and after a certain (random) number of activations disable DCL perk and use ours to start minigame on the next activation.

> But still, I wasn't been able to think up some nice curse to be used in it...

The first thing that came to my mind is to make another mini-game :)
For example, player starts randomly wandering (sandbox package) until a certain number of crits are pressed.

**Reply by kurotatsu77** [reply]
Hehe, interesting idea... I'm mulling over this idea for the Abadon mind break:
1. When player gets orgasms in "losing mind" mode from 100% charged Abadon plug there's a chance to receive Abadon Curse.
2. It applies some kind of tattoo (optional, but nice to have visual indication), but more importantly - effect on the player.
3. Effect is following: while plug is inside, player get some buffs, but when no plug is worn, then player gets debuffs in the same magnitude
4. Effect can have several stages (thinking three maybe?), and can be negated by keeping Abadon Plug in inventory. No debuff if it's close to body! :)
5. Effect can be increased by losing mind while wearing 100% charged Abadon plug
6. Effect decreases over time with each orgasm while no Abadon Plug equipped.
7. This is almost your idea: during no plug orgasm, black goo(and maybe concentrated on higher effects level on some random basis?) is secreted. Here's possibilities to act on it - i.e. try to throw it away, try to collect it, with it being applied on failures or some random chance? Maybe some minigame can be done? Some thoughts needed here...
8. Effect of higher level can require at least two plugs in inventory to disable debuffs, even higher would only accept wearing plug, top-level maybe would force player to wear both of them! Of course magnitudes of both buffs and debuffs should be much higher.

Why I think it's needed? To add effect of true risk to wearing the Plug and allowing it to get full charge and actually fear it. Or just obey the lovely plug and wear them proudly and enjoy buffs :D
This idea is in need for further thinking through and maybe discussion.
I wonder how would @IHateMyKite would feel about this...

**Reply by kurotatsu77** [reply]
Another curse idea could be player randomly losing stamina and magicka while trying to get out of restraints because they become addicted to being bound... This idea even more raw and needs even more thinking through...

**Comment by IHateMyKite** [comment]
Didn't test the latest version yet, but there is still one balance issue which is little throwing off my experience. And that is the added Cursed abadon suit which can be equipped with conc. black goo. The cursed straitjacket is both the extremely easy to escape and also rewards player with biggest amount of gold. It looks to me almost like exploit. This was not that big of an issue in UD as the device could be only equipped once per plug, so it could be seen as kind of secret of additional gold. But in case of US, the device can be equipped several times in row if player is lucky and get big amount of money for almost nothing. Can you per chance remove/edit the suit to balance the issue ?

**Comment by kurotatsu77** [comment]
I respect your vision of the mod and will remove this suit if it really affects negatively the mod in your opinion, it's not hard or complicated thing to do.

But let's discuss this and see if it can be improved/balanced first?

Basic idea in adding this suit was adding high difficulty suit, making escaping Abadon Plug in time much harder. So player can get easy ones like colored catsuits or rope set, medium restrictive set one and this cursed hard one was completing the line-up.  This feels like it all comes together and kind of in harmony for me. 
Gold reward from the suit is essentially irrelevant, because there's exploit of simply wearing 100% strength plug and getting infinite amount of restraints from it, both Abadon (thus making it goo, lockpicks and gold conveyor) and normal ones, which then can be sold to DD vendors and with appropriate perk to any vendor, making ridiculous amount of gold. The latter applies to any device manifest device.

What possible alternatives you see appropriate to removing this suit?
Here's my variants:
1. Make cursed straitjacket harder to escape. It is a cursed item after all. 
2. I've already made sure Abadon Plug is not applied until the quest has been completed. Can do this for the whole suit, so it will drop only after Abadon quest is done. Maybe this will be enough to balance it out?

I've been intending to create discussion about Abadon Plug behavior improvement after getting good stable version out and feature freeze off. Also there are some other things in UD to think on and discuss when the time is good for it.

In the end - I will respect your decision and will act accordingly.

**Reply by IHateMyKite** [reply]
My original vision for this suit/set that it can only be equipped by abadon plug if actor doesn't escape the plug in time. Issue is that at first it was so hard to escape I had to make it almost trivially easy to escape. But that is the point. The whole point of the cursed abadon straitjacket is to not get locked in it, and motivate user to escape the plug as fast as possible. And if they intend to use the plug to farm gold, they will need to get locked in this device and escape it. As I said, I can currently escape the device basically on the first try, its almost comical. And now having option to get locked in the suit even if you didn't had abadon plug before seems to me anti-climatic. Maybe you can just replace the straitjacket with some new device (like Weak Cursed Straitjacket ?). Then you can balance it in you own way, while I can again focus more on making the original straitjacket harder. 

Also regarding the gold, its not really irrelevant. It depends on how hard it is to get the gold. In case of abadon plug, you still have to escape the manifested devices to get the gold, meaning you actually have to invest some time. But in case of cursed straitjacket you almost invest no time, and gets like 8k gold at once. Getting the same amount of gold from normal devices means that you have to escape at least 10 devices, which takes much more time.

Lastly, I was thinking about adding the MCM slider for gold reward multiplier, so users can edit the amount of gold from all devices further.

**Comment by kurotatsu77** [comment]
I think I've got it now! So basically you want Cursed SJ to be extremely punishing device and be used only by final form of the plug.
I can easily copy the current version of it, rename and change it to be more like other straitjackets, albeit slightly tougher and use it instead.

How do you feel about leaving Cursed SJ in the Abadon Execution Potion, as it's intended to be essentially "end of the line" thing?

Do you have similar concerns about rest of the Cursed set? Or I'm good to use it as long as I don't touch Cursed SJ?

Are any other items in UD which should be reserved as quest-only and avoided to be used elsewhere?

**Reply by IHateMyKite** [reply]
- What is Abadon Execution Potion ? Didn't hear about it before
- Other items from the sets are quite basic. Nothing about them is cursed. Only exception is that straitjacket and gas mask (but I don't think there is reason to restrict use of the gas mask, as it's only powerful when combined with SJ)
- Only 2 devices are basically used by the quest: Abadon plug and the Cursed abadon straitjacket. So any other items can be used freely. But this can possibly change in the future.

**Reply by kurotatsu77** [reply]
- It's the extreme potion I've added in US, meant to be alternative to actually execute someone, making them bondage doll instead. It's added to research and Urag's research share quests.
- Wonderful. I'll do as you ask and will make SJ which I use instead of Cursed one, making sure to remove it from everywhere, except Execution Potion, if you green-light it.
- Abadon Plug now can be added along with some suit sets (cursed and random) only after the Abadon Curse quest has been completed, checkbox to suppress it's appearance is in MCM, so user can opt to never receive it as part of suit. May I leave it as it is now or you still want me to change something in this?

**Reply by kurotatsu77** [reply]
Made SJ/catsuit and used it instead of the Cursed one,e xcept for Abadon Execution Potion. Pushed into repo, please have a look and see if you like.
